unit/analyses/variable-sensitivity/abstract_object/index_range.cpp

```diff
@@ -40,14 +40,18 @@ SCENARIO(
     }
   }
 
-  GIVEN("a top constant's range is empty")
+  GIVEN("a top constant's range is has a single nil expression")
```

Comment:
If it tis top then it should be everything so this is definitely an improvement but I'm not quite sure what the nil is supposed to describe.

Reply:
The index_range is only used to during array access, and this supports the existing behaviour. Returning a nil expression here takes us down to https://github.com/jezhiggins/cbmc/blob/b618d2e3703434aa0cfb4d1c991c867b1dea9955/src/analyses/variable-sensitivity/full_array_abstract_object.cpp#L298